core/include/detray/masks/annulus2.hpp

```diff
@@ -12,165 +12,212 @@
+    ///
+    /// @return an intersection status e_inside / e_outside
     DETRAY_HOST_DEVICE intersection::status is_inside(
-        const point2 &p,
-        const scalar t = std::numeric_limits<scalar>::epsilon()) const {
-        // The two quantities to check: r^2 in module system, phi in strips
-        // system
-
-        // In cartesian coordinates go to modules system by shifting origin
-        if constexpr (std::is_same_v<inside_local_t,
-                                     __plugin::cartesian2<detray::scalar>>) {
-            // Calculate radial coordinate in module system:
-            scalar x_mod = p[0] - this->_values[4];
-            scalar y_mod = p[1] - this->_values[5];
-            scalar r_mod2 = x_mod * x_mod + y_mod * y_mod;
-
-            // apply tolerances
-            scalar minR_tol = this->_values[0] - t;
-            scalar maxR_tol = this->_values[1] + t;
-
-            if (r_mod2 < minR_tol * minR_tol or r_mod2 > maxR_tol * maxR_tol)
-                return intersection::status::e_outside;
-
-            scalar phi_strp = getter::phi(p) - this->_values[6];
-            // Check phi boundaries, which are well def. in local frame
-            return (phi_strp >= this->_values[2] - t and
-                    phi_strp <= this->_values[3] + t)
-                       ? intersection::status::e_inside
-                       : intersection::status::e_outside;
-        }
-        // polar strip coordinates given
-        else {
-            // For a point p in local polar coordinates, rotate by avr phi
-            scalar phi_strp = p[1] - this->_values[6];
-
-            // Check phi boundaries, which are well def. in local frame
-            if (phi_strp < this->_values[2] - t ||
-                phi_strp > this->_values[3] + t)
-                return intersection::status::e_outside;
-
-            // Now go to module frame to check r boundaries. Use the origin
-            // shift in polar coordinates for that
-            point2 shift_xy = {-1 * this->_values[4], -1 * this->_values[5]};
-            scalar shift_r = getter::perp(shift_xy);
-            scalar shift_phi = getter::phi(shift_xy);
-
-            scalar r_mod2 = shift_r * shift_r + p[0] * p[0] +
-                            2 * shift_r * p[0] * std::cos(phi_strp - shift_phi);
-
-            // Apply tolerances
-            scalar minR_tol = this->_values[0] - t;
-            scalar maxR_tol = this->_values[1] + t;
-
-            return (r_mod2 >= minR_tol * minR_tol and
-                    r_mod2 <= maxR_tol * maxR_tol)
-                       ? intersection::status::e_inside
-                       : intersection::status::e_outside;
+        const point2_t &loc_p,
+        const scalar_t tol = std::numeric_limits<scalar_t>::epsilon()) const {
+
+        // The two quantities to check: r^2 in disc system, phi in focal system:
+
+        // Rotate by avr phi into the focal system
+        scalar_t phi_strp{loc_p[1] - this->_values[e_average_phi]};
```

Comment:
The ```local_frame_type``` of the annulus is defined as polar now, which corresponds to the disk frame. I followed the ACTS convention of making it the caller's responsibility to give a point in a local coordinate system to the boundary check. But I agree, the current annulus implementation is not stringent yet in correctly defining the coordinate systems (it has multiple), mainly because I am not sure about the exact position of the strips

Reply:
I agree. We can put local position (`point2`) directly into the `is_inside` function.